### PR TITLE
BUG: __eq__ raising for new arrow string dtype for incompatible objects

### LINKED ISSUE #56008: BUG: __eq__ raises NotImplementedError for new arrow string dtype
### Reproducible Example

```python
ser = pd.Series([False, True])
ser2 = pd.Series(["a", "b"], dtype="string[pyarrow_numpy]")
ser == ser2
```

### Issue Description

This shouldn't raise if we want to emulate object dtype behavior

### Expected Behavior

all False return value
cc @jorisvandenbossche thoughts?

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #56008 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by phofl** [comment]
cc @mroeschke gentle ping

File: pandas/core/arrays/string_arrow.py [hunk 1]
```diff
@@ -663,7 +664,10 @@ def _convert_int_dtype(self, result):
         return result
 
     def _cmp_method(self, other, op):
-        result = super()._cmp_method(other, op)
+        try:
```
Comment (mroeschke) [inline pandas/core/arrays/string_arrow.py:667]: I think after https://github.com/pandas-dev/pandas/pull/56003 this won't be needed

Comment (phofl) [reply pandas/core/arrays/string_arrow.py:667]: why? Will this make comparing strings to any non strings less strict?

**Comment by phofl** [comment]
cc @mroeschke

File: pandas/core/arrays/string_arrow.py (hunk 1 above)
Comment (phofl) [reply pandas/core/arrays/string_arrow.py:666]: This is still needed

File: doc/source/whatsnew/v2.2.0.rst [hunk 2]
```diff
@@ -604,6 +604,7 @@ Strings
 - Bug in :meth:`Series.str.find` when ``start < 0`` for :class:`ArrowDtype` with ``pyarrow.string`` (:issue:`56411`)
 - Bug in :meth:`Series.str.replace` when ``n < 0`` for :class:`ArrowDtype` with ``pyarrow.string`` (:issue:`56404`)
 - Bug in :meth:`Series.str.startswith` and :meth:`Series.str.endswith` with arguments of type ``tuple[str, ...]`` for ``string[pyarrow]`` (:issue:`54942`)
+- Bug in comparison operations for ``dtype="string[pyarrow_numpy]"`` raising if dtypes can't be compared (:issue:`56008`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v2.2.0.rst:607]: This is not needed given that's a new dtype in 2.2?

File: pandas/core/arrays/string_arrow.py [hunk 3]
```diff
@@ -662,7 +663,10 @@ def _convert_int_dtype(self, result):
...
+            result = super()._cmp_method(other, op)
+        except pa.lib.ArrowNotImplementedError:
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/string_arrow.py:668]:
```suggestion
        except pa.ArrowNotImplementedError:
```

File: pandas/tests/series/test_logical_ops.py [hunk 4]
```diff
@@ -530,3 +530,19 @@ def test_int_dtype_different_index_not_bool(self):
 
         result = ser1 ^ ser2
         tm.assert_series_equal(result, expected)
+
+    def test_pyarrow_numpy_string_invalid(self):
```
Comment (jorisvandenbossche) [inline pandas/tests/series/test_logical_ops.py:534]: I would expect we already have tests for this behaviour, but then only in cases where it's not yet parametrized to use the string dtype?

File: doc/source/whatsnew/v2.2.0.rst (hunk 2 above)
Comment (phofl) [reply doc/source/whatsnew/v2.2.0.rst:607]: no the dtype was new in 2.1

File: pandas/tests/series/test_logical_ops.py (hunk 4 above)
Comment (phofl) [reply pandas/tests/series/test_logical_ops.py:534]: yes

File: doc/source/whatsnew/v2.2.0.rst (hunk 2 above)
Comment (jorisvandenbossche) [reply doc/source/whatsnew/v2.2.0.rst:607]: Ah, yes, just not the `_numpy` part ;)

**Review by jorisvandenbossche — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Pull request merged**